### PR TITLE
fetch tokens data in nft router indexer

### PR DESCRIPTION
**Pull request by szymcio32**

### CODE REVIEW

File: server/server/transform/process_positions.py [hunk 1]
```diff
@@ -49,39 +43,41 @@ def handle_positions(db: Database, rpc_url: str):
...
+                ))
         processed_positions_records += 1
 
     if positions_update_operations:
         db[Collection.POSITIONS].bulk_write(positions_update_operations)
     logger.info(f'Successfully processed {processed_positions_records} Positions records')
 
 
-def handle_positions_fees(db: Database):
+def handle_positions_fees(db: Database, rpc_url: str):
```
Comment (princearoragithub) [inline server/server/transform/process_positions.py:64]: Here, we are only updating token decimals. Do we even need it? We can get it from positions collection wherever we want. In the API or any further transformation. Just get same position ID from the positions collection.

Comment (princearoragithub) [reply server/server/transform/process_positions.py:64]: From position ID we get tokenAddress, and then from tokens collection we get the decimals.

File: indexer/src/nft_router.ts [hunk 2]
```diff
@@ -57,7 +60,7 @@ import {
     },
   };
     
-  export default function transform({ header, events }: Block) {
+  export default async function transform({ header, events }: Block) {
```
Comment (princearoragithub) [inline indexer/src/nft_router.ts:63]: What is the purpose of adding async here? Should we do it for other functions in indexer?

File: indexer/common/utils.ts [hunk 3]
```diff
@@ -3,6 +3,10 @@ import {
   Transaction
 } from "../common/deps.ts";
 
+export function formatBigIntToString(key: BigInt): string {
```
Comment (princearoragithub) [inline indexer/common/utils.ts:6]: We should not be doing this padding, as we do not have padding anywhere else. It should be similar everywhere, like formatFelt.

File: indexer/src/nft_router.ts (hunk 2 above)
Comment (szymcio32) [reply indexer/src/nft_router.ts:63]: I'm using async call to get tokens from position contract, so the top level function needs to be also async. In factory indexer we don't invoking any async function so it's not needed

File: indexer/common/utils.ts (hunk 3 above)
Comment (szymcio32) [reply indexer/common/utils.ts]: Good point. Thanks

File: server/server/transform/process_positions.py (hunk 1 above)
Comment (szymcio32) [reply server/server/transform/process_positions.py]: Refactored

**Pull request closed without merging**